src/jit/flowgraph.cpp

```diff
@@ -17975,12 +17987,19 @@ void                Compiler::fgSetTreeSeqHelper(GenTreePtr tree)
         break;
     }
 
-    fgSetTreeSeqFinish(tree);
+    fgSetTreeSeqFinish(tree, isLIR);
 }
 
 void
-Compiler::fgSetTreeSeqFinish(GenTreePtr tree)
+Compiler::fgSetTreeSeqFinish(GenTreePtr tree, bool isLIR)
 {
+    // If we are sequencing a node that does not appear in LIR,
+    // do not add it to the list.
+    if (isLIR && ((tree->OperGet() == GT_LIST) || tree->OperGet() == GT_ARGPLACE))
+    {
```

Comment:
I should also point out that `GTK_NOLIR` isn't the sole factor is whether or not a node is well-formed LIR: `GenTree::IsLIR()` contains a few other checks. It does in fact enforce that if a `GT_LIST` or `GT_ARGPLACE` node is encountered in an LIR tree that the node is not linked into the containing block.